server/util.go

```diff
@@ -116,25 +116,25 @@ func bindFlags(basename string, cmd *cobra.Command, v *viper.Viper) (err error)
 // is used to read and parse the application configuration. Command handlers can
 // fetch the server Context to get the CometBFT configuration or to get access
 // to Viper.
-func InterceptConfigsPreRunHandler(cmd *cobra.Command, customAppConfigTemplate string, customAppConfig interface{}, cmtConfig *cmtcfg.Config) error {
+func InterceptConfigsPreRunHandler(cmd *cobra.Command, customAppConfigTemplate string, customAppConfig interface{}, cmtConfig *cmtcfg.Config) (*Context, error) {
```

Comment:
If someone has a better name for this function, we could keep the same behavior, and rename this function to something else. This way it isn't breaking (but increases the API of the SDK).